apps/app_test.go

```diff
@@ -119,3 +119,375 @@ func TestAppId_String(t *testing.T) {
+
+func TestRegistrationIntent_MultipleIntentsViaPortDefinitionIfMultipleContainConsulLabel(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true", "common-tag": "tag"},
+		PortDefinitions: []PortDefinition{
+			PortDefinition{
+				Labels: map[string]string{"consul": "first-name", "first-tag": "tag"},
+			},
+			PortDefinition{
+				Labels: map[string]string{"consul": "second-name", "second-tag": "tag"},
+			},
+		},
+	}
+	task := &Task{
+		Ports: []int{1234, 5678},
+	}
+
+	// when
+	intents := app.RegistrationIntents(task, "-")
+
+	// then
+	assert.Len(t, intents, 2)
+	assert.Equal(t, "first-name", intents[0].Name)
+	assert.Equal(t, 1234, intents[0].Port)
+	assert.Equal(t, []string{"common-tag", "first-tag"}, intents[0].Tags)
+	assert.Equal(t, "second-name", intents[1].Name)
+	assert.Equal(t, 5678, intents[1].Port)
+	assert.Equal(t, []string{"common-tag", "second-tag"}, intents[1].Tags)
+}
+
+func TestHasSameConsulNamesAs_SameConfigsWithoutPortDefinitions(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true"},
+	}
+
+	// expect
+	assert.True(t, app.HasSameConsulNamesAs(app))
+}
+
+func TestHasSameConsulNamesAs_DifferentConfigsSameNameWithoutPortDefinitions(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true"},
+	}
+	other := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "app-name"},
+	}
+
+	// expect
+	assert.True(t, app.HasSameConsulNamesAs(other))
+}
+
+func TestHasSameConsulNamesAs_DifferentConfigsWithoutPortDefinitions(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "some-name"},
+	}
+	other := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "different-name"},
+	}
+
+	// expect
+	assert.False(t, app.HasSameConsulNamesAs(other))
+}
+
+func TestHasSameConsulNamesAs_SameConfigsWithPortDefinitions(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true"},
+		PortDefinitions: []PortDefinition{
+			PortDefinition{
+				Labels: map[string]string{"consul": "true"},
+			},
+		},
+	}
+
+	// expect
+	assert.True(t, app.HasSameConsulNamesAs(app))
+}
+
+func TestHasSameConsulNamesAs_DifferentConfigsSameNamesWithPortDefinitions(t *testing.T) {
+	t.Parallel()
+
+	// given
+	app := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true"},
+		PortDefinitions: []PortDefinition{
+			PortDefinition{
+				Labels: map[string]string{"consul": "true"},
+			},
+		},
+	}
+	other := &App{
+		ID:     "app-name",
+		Labels: map[string]string{"consul": "true"},
+		PortDefinitions: []PortDefinition{
+			PortDefinition{
+				Labels: map[string]string{"consul": "app-name"},
+			},
+		},
+	}
+
+	// expect
+	assert.True(t, app.HasSameConsulNamesAs(other))
```

Comment:
the opposite way assertion `other.HasSameConsulNamesAs(app))` would be welcome :)